Tasks/Gradle/gradletask.ts

```diff
@@ -94,43 +96,52 @@ if (specifiedJavaHome) {
     tl.debug('Set JAVA_HOME to ' + specifiedJavaHome);
     process.env['JAVA_HOME'] = specifiedJavaHome;
 }
+ 
+/* Trigger code coverage enable flow and then build */
+enableCodeCoverage()
+    .then(function (resp) {
+        tl.debug("Enabled code coverage successfully: " + reportingTaskName);
+        gb.arg(reportingTaskName);
+    }).catch(function (err) {
+        tl.warning("Failed to enable code coverage: " + err);
+    }).fin(function () {
+        setGradleOpts();
+        enableSonarQubeAnalysis();
+        execBuild();
+    });
 
-if (isCodeCoverageOpted) {
-    tl.debug("Option to enable code coverage was selected and is being applied.");
-    enableCodeCoverage();
+/* Actual execution of Build and further flows*/
+function execBuild() {
+    var gradleResult;
+    gb.exec()
+        .then(function (code) {
+            gradleResult = code;
+            publishTestResults(publishJUnitResults, testResultsFiles);
+            publishCodeCoverage(isCodeCoverageOpted);
+            return processCodeAnalysisResults();
+        })
+        .then(() => {
+            tl.exit(gradleResult);
+        })
+        .fail(function (err) {
+            publishTestResults(publishJUnitResults, testResultsFiles);
+            console.error(err);
+            tl.debug('taskRunner fail');
+            tl.exit(1);
+        });
 }
 
-if (isSonarQubeEnabled) {
-    // Looks like: 'SonarQube analysis is enabled.'
-    console.log(tl.loc('codeAnalysis_ToolIsEnabled'), sqCommon.toolName);
+function enableSonarQubeAnalysis() {
+    if (isSonarQubeEnabled) {
+        // Looks like: 'SonarQube analysis is enabled.'
+        console.log(tl.loc('codeAnalysis_ToolIsEnabled'), sqCommon.toolName);
 
-    gb = sqGradle.applyEnabledSonarQubeArguments(gb);
-    gb = sqGradle.applySonarQubeCodeCoverageArguments(gb, isCodeCoverageOpted, ccTool, summaryFile);
+        gb = sqGradle.applyEnabledSonarQubeArguments(gb);
+        gb = sqGradle.applySonarQubeCodeCoverageArguments(gb, isCodeCoverageOpted, ccTool, summaryFile);
```

Comment:
SonarQube needs to consume the test results and the code coverage details. With the previosu implementation of CC, the report was set is a non-default location. Is there still a hard requirement to do this? Note that we'd like to remove SonarQube from the Maven/Gradle tasks and place it in its own separate task. However, we do not know where the code coverage file will be droppped! - the simplest solution is to let it go to the default location.